crates/nu-command/src/filters/roll/roll_down.rs

```diff
@@ -62,10 +62,17 @@ impl Command for RollDown {
         input: PipelineData,
     ) -> Result<nu_protocol::PipelineData, ShellError> {
         let by: Option<usize> = call.get_flag(engine_state, stack, "by")?;
+        let ctrlc = engine_state.ctrlc.clone();
+        let metadata = input.metadata();
+
         let value = input.into_value(call.head);
         let rotated_value = vertical_rotate_value(value, by, VerticalDirection::Down)?;
 
-        Ok(rotated_value.into_pipeline_data())
```

Comment:
Can you use `Ok(rotated_value.into_pipeline_data().set_metadata(metadata))` instead of the steps below?

Reply:
I already try to this method, but, `vertical_rotate_value` and `horizontal_rotate_value` function returns `Value::List {}`.  Even if the `set_metadata` function is called, the value inside `PipelineData` is not a `ListStream`, so `ls_colors` does not seem to be output normally.
This is why I called `as_list`.

Reply:
Ahh, thanks for the explanation. I've added the ability to also use metadata when working with `List` in https://github.com/nushell/nushell/pull/4635.

This should let you just pass the metadata through without having to do extra steps.

Reply:
Oh! Cool 😄  thanks! I updated the your feedback.